dali/operators/audio/nonsilence_op.cc

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/core/static_switch.h"
+#include "dali/operators/audio/nonsilence_op.h"
+#include "dali/pipeline/data/views.h"
+
+namespace dali {
+
+DALI_SCHEMA(NonsilenceRegion)
+                .DocStr(R"code(The operator performs leading and trailing silence detection in an audio buffer.
+This operators' behaviour can be described as::
+
+  def nonsilence(buffer, cutoff_value, top_db, ref_db):
+      buffer = ToDecibels(buffer, top_db, ref_db)
+      begin = end = 0
+      for i in [0 .. len(buffer)):
+          if buffer[i] > cutoff_value:
+              begin = i; break
+      for i in [len(buffer)-1 .. begin]:
```

Comment:
```suggestion
      for i in range(len(buffer)-1, begin, -1):
```

Reply:
I didn't want to be python-precise here, cause somebody might come and say "hey, this is not accurate and this snippet in python doesn't work". 
Plus, this is not **precisely** what we do in nonsilence. If somebody takes this snippet, in specific argument setups, results will be different.
 On top of that, using `range` in python for loops is lately quite mocked by the community and shows bad python skills ;)

Instead I went for pythonish pseudo code, to leave no expectations, that following snippet will actually work.

Reply:
> I didn't want to be python-precise here, cause somebody might come and say "hey, this is not accurate and this snippet in python doesn't work".

Ok

> Plus, this is not precisely what we do in nonsilence. If somebody takes this snippet, in specific argument setups, results will be different.

If you use pseudo-code then it should do precisely what it suppose to. Otherwise, people will be surprised by the effect. Pseudocode is to make things simpler, not less accurate.
The description should explain that the average inside the moving window is calculated and then the threshold is applied. Without it, the user has no idea what `window_length` is for.

> On top of that, using a range in python for loops is lately quite mocked by the community and shows bad python skills ;)

You want to get index here, so this is rater the case when it is allowed.

Reply:
You're right. I removed the pseudo-code part. I guess even this one-sentence explanation makes do for describing this operator